container/sagemaker/tensorflow-serving.js

```diff
@@ -168,38 +168,21 @@ function json_lines_request(r, data) {
 function csv_request(r) {
     var data = r.requestBody
 
-    // look for initial quote or numeric-only data in 1st field
-    var needs_quotes = data.search(/^\s*("|[\d.Ee+\-]+\s*,)/) != 0
-
     var lines = data.trim().split(/\r?\n/)
 
-    // start instances json - omit outer list for single example
-    var json = '{"instances":'
-    if (lines.length != 1) {
-        json += '['
-    }
+    var builder = []
+    builder.push('{"instances":[')
 
     for (var i = 0; i < lines.length; i++) {
         var line = lines[i].trim()
         if (line) {
             var instance = (i == 0) ? '[' : ',['
-
-            if (needs_quotes) {
-                instance += '"'
-                instance += line.replace(',', '","')
-                instance += '"'
-            } else {
-                instance += line
-            }
-
-            instance += ']'
-
-            json += instance
+            builder.push(instance)
```

Comment:
the quoting functionality has been deleted completely. this is a regression that will cause problems for string input data.

Reply:
ah, right, for some reason I thought this wouldn't be needed any more, but I just realized I was testing this with string data already in quotes. i'll re-add this so that unquoted, non-numeric string data is quoted correctly in the "instances" JSON

Reply:
I don't think the quoting functionality works anyway. "iris,0.5,1.0,1.5" gives `{"instances":["iris","0.5,1.0,1.5"]}` instead of `{"instances":["iris", 0.5, 1.0, 1.5]}`

Reply:
Deleting the quote functionality also removes the quotes of "iris" in the example given above. Maybe we should differentiate digits and alphabets?

Reply:
Yeah, I think that's what we'd have to do -- it definitely can't be removed either, but I think we'd have to parse each line one CSV entry at a time. I'd like to do this in Python instead to use efficient libraries and do it faster and maybe allow for larger payloads.

I also want to float the idea of building and distributing a separate image for Batch jobs, since I suppose the reason the `js_include` solution was chosen was to optimize on latency for small payloads, but at the cost of payload size and throughput.

Reply:
@andremoeller 

the existing code is meant to do conversions like this (csv left, json right)

```
word1,word2 and word3,word4 -> ["word1","word2 and word3","word4"]
"word1","word2 and word3","word4" -> ["word1","word2 and word3","word4"]
"word1,word2 and word3,word4" -> ["word1,word2 and word3,word4"] # note one field
```
for this input: `"iris,0.5,1.0,1.5"` i would have expected `""iris,0.5,1.0,1.5"` since the quotes in the csv mean it's only one column. so that looks like a bug in the original code -- the `line.replace(',', '","')` is only replacing the first comma because the replacement expressions is a string instead of a regex. oops. we should correct that in this pr after restoring the quoting code.

as for implementing this in python instead... let's not go there. i don't think your performance assumptions are correct, especially since this would mean every request would have to go through the python service. we've already built containers like that and we know this one is faster. even though we are adding python support in another PR, the intention there is that that is an optional feature for users who require custom code in the request flow. We expect many users to continue using this container without custom code, and in those cases, we won't even launch the python service.

Reply:
OK, I readded the string behavior and made the change you suggested (regex instead of simple string replace), and got the expected results for those cases. But I think Chuyang's right that we'll basically have to go through each value and see whether it's numeric to decide whether to quote it, or we end up with quoted floats and ints. I'll see what this looks like, then rerun performance tests (and maybe add more) and see whether the "stringbuilder" does OK relative to concatenation.

About Python: OK, as long as we can get good performance (where performance doesn't mean "fast" like low latency, but "fast" like high throughput) on payloads like 100MB, or 10MB and 10 concurrent requests, 20MB and 5 concurrent requests, 6MB and 16 concurrent requests, etc. this isn't needed. We seem pretty far away from that at this point (even this PR doesn't bring us to 6MB and 16 concurrent requests), but there might be simpler solutions to this.